bitcoin/src/consensus/encode.rs

```diff
@@ -936,10 +936,20 @@ mod tests {
         assert_eq!(serialize(&cd), vec![5, 0, 0, 0, 162, 107, 175, 90, 1, 2, 3, 4, 5]);
     }
 
+    fn prefix_varint_len(data: &Vec<u8>) -> Vec<u8> {
+        // Takes a reference to a Vec<u8> as input and prefixes the length of the vector as a VarInt.
+        let len = VarInt(data.len() as u64);
+        let len_bytes = len.0.to_le_bytes();
+        let mut prefixed_data = Vec::new();
+        prefixed_data.append(&mut len_bytes[0..len.len()].to_vec());
```

Comment:
This looks very wrong, it should be `len.consensus_encode(&mut prefixed_data).unwrap();`

Reply:
Ah !! I just realized while I was looking for `consensus_encode` that in the existing `serialize` function already does what I want to do. This won't add any good case to test. It is same thing.

> I'd actually prefer to remove the impl on Vec completely and explicitly use some function or a newtype which prefixes it with varint length.

Can you please explain this again ?

Reply:
Delete `impl<T: Encodable> Encodable for Vec<T>` and decodable counterpart. Then make a function `fn encode_len_prefixed<T: Encodable>(writer: Write, slice: &[T]) -> io::Result<()>` and its decode counterpart. Then at all places compiler complains about missing impl call that new function instead.